client/my-sites/marketing/controller.js

```diff
@@ -41,7 +41,11 @@ export const redirectDefaultConnectionsDomain = async ( context ) => {
 
 	let recentSiteSlug = getSiteSlug( state, recentSiteId );
 	if ( ! recentSiteSlug ) {
-		await dispatch( requestSite( recentSiteId ) );
+		try {
+			await dispatch( requestSite( recentSiteId ) );
+		} catch {
+			// proceed despite a failed site request
```

Comment:
What's the idea here? To use the previously cached site data (if any)?

Reply:
The idea is that the calling code doesn't check the return value of `requestSite`. Instead, it detects success or failure by looking at its side effects, i.e., what it wrote to state. Something like:
```js
if ( ! getSite( getState(), siteId ) ) {
  await dispatch( requestSite( siteId ) );
  if ( ! getSite( getState(), siteId ) ) {
    console.error( 'the site is not there even after requesting it, means the request failed' );
  }
}
```
The `.catch( () => null )` blocks in other parts of this patch express the same idea.